my-app/src/app/modules/courses-page/courses-page.component.ts

```diff
@@ -16,18 +13,28 @@ import {FilterPipe} from "./core";
   providers: [FilterPipe]
 })
 export class CoursesPageComponent implements OnInit {
-  public courses: Course[] | undefined;
+  public initialCourses: Course[] | undefined;
+  public currentCourses: Course[] | undefined;
   public searchValue: string | undefined;
 
-  constructor(private filter: FilterPipe) {
+  constructor(private filterPipe: FilterPipe,
+              private coursesHandlerService: CoursesHandlerService) {
   }
 
   ngOnInit(): void {
-    this.courses = COURSE_DATA;
+    this.initialCourses = this.coursesHandlerService.getList();
+    this.currentCourses = this.initialCourses;
   }
 
   public onDeleteCourse(courseId: number): void {
-    console.log(courseId);
+    if (confirm(
+        `Are you sure to delete '${this.initialCourses?.find(item => item.id === courseId)?.title}' course`) &&
```

Comment:
I would extract `this.initialCourses?.find(item => item.id === courseId)?.title` into a const for readability.

Reply:
Fixed, thank you